sentry-core/src/main/java/io/sentry/core/Dsn.java

```diff
@@ -48,20 +48,20 @@ public Dsn(String dsn) throws InvalidDsnException {
     try {
       URI uri = new URI(dsn);
       String userInfo = uri.getUserInfo();
-      if (userInfo == null || userInfo.length() == 0) {
+      if (userInfo == null || userInfo.isEmpty()) {
         throw new IllegalArgumentException("Invalid DSN: No public key provided.");
       }
       String[] keys = userInfo.split(":");
       publicKey = keys[0]; // TODO: test lack of delimiter returns whole value as first index
-      if (publicKey == null || publicKey.length() == 0) {
+      if (publicKey == null || publicKey.isEmpty()) {
         throw new IllegalArgumentException("Invalid DSN: No public key provided.");
       }
       secretKey = keys.length > 1 ? keys[1] : null;
       String uriPath = uri.getPath();
       int projectIdStart = uriPath.lastIndexOf("/") + 1;
       path = uriPath.substring(0, projectIdStart);
       projectId = uriPath.substring(projectIdStart);
-      if (projectId == null || projectId.length() == 0) {
+      if (projectId.isEmpty()) {
```

Comment:
It can never be null?

Reply:
no, otherwise will throw NPE or something else on the line above